exec/java-exec/src/test/java/org/apache/drill/exec/expr/fn/impl/TestStringFunctions.java

```diff
@@ -114,6 +114,19 @@ public void testRegexpMatches() throws Exception {
   }
 
   @Test
+  public void testRegexpReplace() throws Exception {
+    testBuilder()
+        .sqlQuery("select regexp_replace(a, 'a|c', 'x') res1, regexp_replace(b, 'd', 'zzz') res2 " +
+                  "from (values('abc', 'bcd'), ('bcd', 'abc')) as t(a,b)")
+        .unOrdered()
+        .baselineColumns("res1", "res2")
+        .baselineValues("xbx", "bczzz")
+        .baselineValues("bxd", "abc")
+        .build()
+        .run();
+  }
+
```

Comment:
sure sounds good. I'll do that.

On Mon, Jun 6, 2016 at 8:50 PM, Jinfeng Ni <notifications@github.com> wrote:

> In
> exec/java-exec/src/test/java/org/apache/drill/exec/expr/fn/impl/TestStringFunctions.java
> <https://github.com/apache/drill/pull/512#discussion_r65994802>:
>
> > @@ -114,6 +114,19 @@ public void testRegexpMatches() throws Exception {
> >    }
> >
> >    @Test
> > +  public void testRegexpReplace() throws Exception {
> > +    testBuilder()
> > +        .sqlQuery("select regexp_replace(a, 'a|c', 'x') res1, regexp_replace(b, 'd', 'zzz') res2 " +
> > +                  "from (values('abc', 'bcd'), ('bcd', 'abc')) as t(a,b)")
> > +        .unOrdered()
> > +        .baselineColumns("res1", "res2")
> > +        .baselineValues("xbx", "bczzz")
> > +        .baselineValues("bxd", "abc")
> > +        .build()
> > +        .run();
> > +  }
> > +
>
> Can you add at least one case for the non-ASCII case, since one of the
> issue this PR tries to address is the incorrect result for regex function
> over non-ASCII input?
>
> A new unit test case will help verify this PR fix the problem, and make
> sure any new change will not regress this functionality.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/drill/pull/512/files/84846241bf8beef2a090414aacda2991911022d7#r65994802>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJoEwm0Aua6mNJEOorpLJBjgTV27LhgUks5qJMBTgaJpZM4IrlUy>
> .
>